metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/RestHighLevelClientFactory.java

```diff
@@ -89,6 +91,35 @@ private static RestClientBuilder loadRestHttpClient(@Nonnull String host, int po
 
     return builder;
   }
+  
+  @Nonnull
+  private static RestClientBuilder loadRestHttpClient(@Nonnull String host, int port, String pathPrefix, int threadCount,
+      int connectionRequestTimeout, String username, String password) {
+    RestClientBuilder builder = RestClient.builder(new HttpHost(host, port, "http"))
+        .setHttpClientConfigCallback(httpAsyncClientBuilder -> httpAsyncClientBuilder
+            .setDefaultIOReactorConfig(IOReactorConfig.custom().setIoThreadCount(threadCount).build()));
+
+    if (!StringUtils.isEmpty(pathPrefix)) {
+      builder.setPathPrefix(pathPrefix);
+    }
+    
+    builder.setHttpClientConfigCallback(new RestClientBuilder.HttpClientConfigCallback() {
```

Comment:
Seems like this is the only thing being added to the original one? Can we call the original function and then set the client config callback of the returned RestClientBuilder?

Reply:
yup make, sense